include/boost/utility/string_view.hpp

```diff
@@ -135,6 +144,12 @@ namespace boost {
         BOOST_CONSTEXPR const_reference back()  const                { return ptr_[len_-1]; }
         BOOST_CONSTEXPR const_pointer data()    const BOOST_NOEXCEPT { return ptr_; }
 
+#ifndef BOOST_NO_CXX17_HDR_STRING_VIEW
+        operator std::basic_string_view<charT, traits>() const {
```

Comment:
should this be `noexcept`?

Reply:
Good catch, it really should be.

This whole PR likely needs a lot of cleanup and fixes to get back into a good state

Reply:
while we are at it: it can be constexpr too. Also the closing brace is misindented